### PR TITLE
Alerting: migration to fail Graphite query contains unexpanded references.

### PR DESCRIPTION
**Pull request by yuri-tceretian**
**What this PR does / why we need it**:
Graphite data source plugin supports nested references between queries. For example, a panel can use many queries that refer to other queries:
```
A: sumSeries(some.query)
B: asPercent( diffSeries( #A, timeShift(#A, '1d') ) ,sumSeries( #A ) )
```
In this example, query `B` refers to query `A`.

The Graphite plugin expands the query `B` to the following:

`asPercent( diffSeries( sumSeries(some.query), timeShift(sumSeries(some.query), '1d') ) ,sumSeries( sumSeries(some.query)) )` 

and saves it to a field `targetFull`. Since https://github.com/grafana/grafana/pull/42493 alerting migration uses that field if it exists. However, due to a bug https://github.com/grafana/grafana/issues/42623 sometimes the expanded query still contains references to other queries, and this causes migrated alert rules to start failing.

This PR updates the migration code to check the field `target` (and `targetFull`) for unexpanded queries and fail if there is at least one. This is only applicable to Graphite data sources.

Also, PR updates intermediate structure alertRule to keep field `datasourceType`

### CODE REVIEW

**Review by JacobsonMT — APPROVED** [review]
LGTM :+1:

File: pkg/services/sqlstore/migrations/ualert/alert_rule.go [hunk 1]
```diff
@@ -171,9 +177,30 @@ func migrateAlertRuleQueries(data []alertQuery) ([]alertQuery, error) {
 	return result, nil
 }
 
-// fixGraphiteReferencedSubQueries attempts to fix graphite referenced sub queries, given unified alerting does not support this.
-// targetFull of Graphite data source contains the expanded version of field 'target', so let's copy that.
-func fixGraphiteReferencedSubQueries(queryData map[string]json.RawMessage) map[string]json.RawMessage {
+// verifyQuery creates a validation function that checks whether the
```
Comment (gillesdemey) [inline pkg/services/sqlstore/migrations/ualert/alert_rule.go:180]: Perhaps the diff is a bit weird bit it seems like the sentence is cut off here 🤔

Comment (yuri-tceretian) [reply pkg/services/sqlstore/migrations/ualert/alert_rule.go:180]: lol

**Comment by yuri-tceretian** [comment]
Closing this PR for now because it seems that it does not really bring any improvements in terms of migration from legacy to a new alerting. The reason for that is that if the legacy rule is configured with a broken query it would not work, and therefore migration of that rule to the same non-working state does not break anything. 
The broken query is a purely hypothetical situation because we have not found any evidence that that really happens. We will reopen this PR if evidence emerges.

**Pull request closed without merging**